lsass/server/auth-providers/ad-open-provider/join/join.c

```diff
@@ -732,7 +732,7 @@ LsaJoinDomainInternal(
                sizeof(pwszMachinePassword)/sizeof(pwszMachinePassword[0]));
     BAIL_ON_LSA_ERROR(dwError);
 
-    if (pwszMachinePassword[0] == '\0')
+    if (pwszMachinePassword[0] == NULL || *pwszMachinePassword[0] == '\0')
```

Comment:
There is a really big problem with pwszMachinePassword. This is declared as the wrong data type: 
      PWSTR pwszMachinePassword[MACHPASS_LEN+1] = {0};
This declaration is wrong. This allocating an array of PWSTR pointers of MACHPASS+1 length; an extra level of indirection in this declaration. This really should be:
    WSTR pwszMachinePassword[MACHPASS_LEN+1] = {0};
This is the correct variable declaration for the use in LsaJoinDomainInternal().

The call to LsaGenerateMachinePassword() has an explicit cast to (PWSTR) because pwszMachinePassword is declared as the wrong type. This call is being passed a PWSTR * variable, because pwszMachineAccount is declared as the wrong type.

Kindly fix this declaration, and make the appropriate changes to clean up this incorrectly declared variable.

Reply:
Sorry for the delayed response! (I was expecting a Github notification email upon getting review comments on this pull request, but I got none. So I had assumed that this was still under review, and only accidentally noticed today that you had reviewed it long ago!).

Thank you for the detailed analysis! I have updated the fix with your suggested changes and sent a new pull request (https://github.com/vmware/likewise-open/pull/13)